TaskVlopper.Repository/UserTaskAssignmentRepository.cs

```diff
@@ -12,37 +12,58 @@ public class UserTaskAssignmentRepository : BaseRepository<UserTaskAssignment>,
     {
         public IEnumerable<UserTaskAssignment> GetTaskAssignmentByTaskId(int taskId)
         {
-            return this.GetAll().Where(x => x.TaskID == taskId);
+            return GetTaskAssignmentByTaskIdQueryable(taskId).AsEnumerable();
         }
 
         public IEnumerable<UserTaskAssignment> GetTaskAssignmentByUserId(string userId)
         {
-            return this.GetAll().Where(x => x.UserID == userId);
+            return GetTaskAssignmentByUserIdQueryable(userId).AsEnumerable();
         }
 
         public IEnumerable<UserTaskAssignment> GetTaskAssignmentByUserIdAndTaskId(string userId, int taskId)
         {
-            return this.GetAll().Where(x => x.TaskID == taskId && x.UserID == userId);
+            return GetTaskAssignmentByUserIdAndTaskId(userId, taskId).AsEnumerable();
         }
 
         public UserTaskAssignment GetTaskAssignmentByUserIdAndProjectIdAndTaskId(string userId, int projectId, int taskId)
         {
-            return this.GetTaskAssignmentByUserIdAndProjectIdAndTaskIdQueryable(userId, projectId, taskId).Single();
+            return this.GetAll().Where(x => x.ProjectID == projectId && x.TaskID == taskId && x.UserID == userId).Single();
```

Comment:
I'd also use queryable here as execution and then Single() on it. It's Queryable anyway